Feature/multiple equal

pawanrawal opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dgraph-io/dgraph/996)
<!-- Reviewable:end -->

manishrjain commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[gql/parser.go, line 1394 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSEoeRbM3W5AkI6vnw:-KlSEoeRbM3W5AkI6vnx:b-1ex4r) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/gql/parser.go#L1394)):*
> ```Go
> 					expectLang = false
> 				} else {
> 					// We only want to parse tokens for inequality functions and not for Geo functions.
> ```

100 chars.

---

*[gql/parser.go, line 2295 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSH57OvMYNsDDfWPwR:-KlSH57OvMYNsDDfWPwS:b-gfd9ga) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/gql/parser.go#L2295)):*
> ```Go
> }
> 
> func isInequality(name string) bool {
> ```

This doesn't belong to gql. it should be done where we handle geo stuff, etc.

---

*[worker/task.go, line 597 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIaj6X09IEQutdDcu:-KlSIaj6X09IEQutdDcv:b-nhkywq) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L597)):*
> ```Go
> )
> 
> func ensureArgsCount(funcStr []string, expected int, op string) error {
> ```

No need for op.

---

*[worker/task.go, line 605 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSHUFyZpCj4DxMsmjP:-KlSHUFyZpCj4DxMsmjQ:b-yieaql) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L605)):*
> ```Go
> 				funcStr[0], expected, actual, funcStr[2:])
> 		}
> 	case "eq":
> ```

case eq:

---

*[worker/task.go, line 638 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSHVkzec0nZ6JbMc8a:-KlSHVkzec0nZ6JbMc8b:b-ntsqnm) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L638)):*
> ```Go
> 	case CompareAttrFn:
> 		// Only eq can have multiple args. It should have atleast one.
> 		if fc.fname == "eq" {
> ```

Use consts

---

*[worker/task.go, line 645 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIN-gBUb_ND3iCBXL:-KlSIN-gBUb_ND3iCBXM:b-an0z0q) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L645)):*
> ```Go
> 		} else {
> 			// Others can have only 1 arg.
> 			err = ensureArgsCount(q.SrcFunc, 1, eq)
> ```

if len(q.SrcFunc) < 3 { return errors.Errorf(..) }

---

*[worker/task.go, line 707 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIZVHhCeVHphLPLg7:-KlSIZVHhCeVHphLPLg8:b-p1kgh9) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L707)):*
> ```Go
> 		// srcfunc 0th val is func name and and [2:] are args.
> 		// we tokenize the arguments of the query.
> 		err = ensureArgsCount(q.SrcFunc, 1, eq)
> ```

if err := ensure; err != nil { .. }

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/996)*
<!-- Sent from Reviewable.io -->

pawanrawal commented:
Review status: 1 of 5 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[worker/task.go, line 597 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIaj6X09IEQutdDcu:-KlSKZCAjRmzNvJjTxM0:b-hfpv0l) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L597)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

No need for op.
</blockquote></details>

removed

---

*[worker/task.go, line 605 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSHUFyZpCj4DxMsmjP:-KlSKWvq8VTwBT3Emnew:b-hfpv0l) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L605)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

case eq:
</blockquote></details>

removed

---

*[worker/task.go, line 638 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSHVkzec0nZ6JbMc8a:-KlSKe9rttORrxinkjKX:bb6un1j) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L638)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

Use consts
</blockquote></details>

Done

---

*[worker/task.go, line 645 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIN-gBUb_ND3iCBXL:-KlSKgZBpFwRUbMgqatb:befm6ws) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L645)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

if len(q.SrcFunc) < 3 { return errors.Errorf(..) }
</blockquote></details>

Done for eq

---

*[worker/task.go, line 707 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSIZVHhCeVHphLPLg7:-KlSKoq48O_22qnWVNOW:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/worker/task.go#L707)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

if err := ensure; err != nil { .. }
</blockquote></details>

Done.

---

*[gql/parser.go, line 1394 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSEoeRbM3W5AkI6vnw:-KlTFbXdUf0HGtqBTnmK:b-hfpv0l) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/gql/parser.go#L1394)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

100 chars.
</blockquote></details>

removed

---

*[gql/parser.go, line 2295 at r1](https://reviewable.io:443/reviews/dgraph-io/dgraph/996#-KlSH57OvMYNsDDfWPwR:-KlTFq92SlS68IBjS30j:b-jh7gox) ([raw file](https://github.com/dgraph-io/dgraph/blob/814b331096c9cd91be4b1037702e55a2054e9a71/gql/parser.go#L2295)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

This doesn't belong to gql. it should be done where we handle geo stuff, etc.
</blockquote></details>

We handle geo stuff in task.go. Handling this there.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dgraph-io/dgraph/996)*
<!-- Sent from Reviewable.io -->

pawanrawal commented:
Opened another PR.
